docs/dx.rst

```diff
@@ -0,0 +1,187 @@
+.. _dx:
+
+DNAnexus
+========
+
+Canonical Paths on DNAnexus
+---------------------------
+Every file resource on DNAnexus is given a canonical ID to refer it by. This is
```

Comment:
Very first: explain stor format for paths:

```
DNAnexus paths on stor are prefixed with dx:// and have two components:
dx://<PROJECT>:<FILE_OR_FOLDER>

where project and file can be virtual paths (i.e., human names) or canonical paths (opaque globally unique IDs that the platform assigns) - see <link to DX API> for more details.
```
1 thing is explaining how virtual_path and canonical_path methods work
```
>>> stor.Path('dx://Myelin_Build_Test').canonical_path
DXCanonicalPath('dx://project-....')
```